pyproject.toml

```diff
@@ -22,8 +22,14 @@ classifiers = [
 
 [tool.poetry.dependencies]
 python = "^3.8"
-
-jsonschema = "*"
+# Pinning jsonschema because its dependencies introduced
+# a dependency of rustc, which we don't want to have installed
+# in charms that use SDI
+# For more information go to canonical/bundle-kubeflow/issues/648
+# For more reference go to:
+# https://github.com/python-jsonschema/jsonschema/issues/1117
+# https://github.com/python-jsonschema/jsonschema/issues/1114
+jsonschema = "^4.17.3"
```

Comment:
if the intent is to prevent 4.18.x builds  I think this should be updatd according to [wildcard requirements](https://python-poetry.org/docs/dependency-specification/#wildcard-requirements)

```suggestion
jsonschema = "4.17.*"
```

Reply:
I vote for something [a little wider](https://github.com/canonical/serialized-data-interface/pull/49#discussion_r1273538337) like `jsonschema = ">4, <4.18"` just in case someone else pinned a bit more tightly, but I'm cool with this too!

Reply:
Sure! i'm fine with going wider.